Chore: Addon documentation restructure (#13358)
storybookjs/storybook: docs/addons/addon-types.md

---
title: 'Types of addons'
---

Each Storybook addon is classified into two general categories, UI-based or Presets. Documented here are some of the features available for each type of addon. Use it as a reference when creating your next Storybook addon.

## UI-based addons

This particular type of addons allows you to customize Storybook's UI. If the addon you're developing falls into this category you can use some of the following elements to further customize the interface.

> ### Panels

Reviewer: Could you add images of these UI elements from @domyen's post? https://storybook.js.org/blog/addon-catalog-sneak-peak/

Author: @winkerVSbecks probably better to use an image that matches the code snippet for accuracy. I'm going to commit them shortly.

Author: Images are added with the latest commits.

---
title: 'Types of addons'
---

> Each Storybook addon is classified into two general categories, UI-based or Presets. Documented here are some of the features available for each type of addon. Use it as a reference when creating your next Storybook addon.

Reviewer 2: ```suggestion
Each Storybook addon is classified into two general categories, UI-based or Presets. Each type of addons features are documented here. Use this as a reference when creating your addon.
```

Author: Fixed with the latest commit

## UI-based addons

> This particular type of addons allows you to customize Storybook's UI. If the addon you're developing falls into this category you can use some of the following elements to further customize the interface.

Reviewer 2: ```suggestion
UI-based addons allow you to customize Storybook's UI using the following elements.
```

Edit for conciseness. This can be shortened into one sentence. 

Also be judicious about "some", that word is imprecise and doesn't give the reader confidence in the docs.

### Panels

> This type of UI element allows you to add your own custom <code>Panel</code> to Storybook's UI. Currently the most common use case available throughout our ecosystem. Good examples that use this type of interface are: [@storybook/actions](../essentials/actions.md) and [@storybook/a11y](https://github.com/storybookjs/storybook/tree/next/addons/a11y).

Reviewer 2: ```suggestion
Panel addons allow you to add your own UI in Storybook's addon panel. This is the most common type of addon in the ecosystem. For example, the official [@storybook/actions](../essentials/actions.md) and [@storybook/a11y](https://github.com/storybookjs/storybook/tree/next/addons/a11y) use this pattern.
```

Author: Fixed with the latest commit

![Storybook panel](./storybook-panel.png)

> Below is the boilerplate code that you can use in your own addon to add a `Panel` in Storybook's UI.

Reviewer 2: Rephrase this doc to describe what the reader should do in fewer words. Get straight to the point.

For example:
"Below is the boilerplate code that you can use in your own addon to add a `Panel` in Storybook's UI."

"Add a `Panel` to Storybook's UI using this boilerplate code."

Author: Fixed with the latest commit

<!-- prettier-ignore-start -->

<CodeSnippets
  paths={[
    'common/storybook-addon-panel-example.js.mdx',
  ]}
/>

<!-- prettier-ignore-end -->

### Toolbars

Toolbar addons allow you to add your own custom tools in Storybook's Toolbar. For example the official [@storybook/backgrounds](../essentials/backgrounds.md) and the [storybook-addon-outline](https://github.com/chromaui/storybook-outline) use this pattern.

![Storybook toolbar addon](./storybook-toolbar.png)

Use this boilerplate code to add a new `button` to Storybook's Toolbar:

<!-- prettier-ignore-start -->

<CodeSnippets
  paths={[
    'common/storybook-addon-toolbar-example.js.mdx',
  ]}
/>

<!-- prettier-ignore-end -->

### Tabs

Tab addons allow you to create your own custom tabs in Storybook. For example the official [@storybook/addon-docs](../writing-docs/introduction.md) uses this pattern.

![Storybook tab addon](./storybook-tab.png)

Use this boilerplate code to add a new `Tab` to Storybook's UI:

<!-- prettier-ignore-start -->

<CodeSnippets
  paths={[
    'common/storybook-addon-tab-example.js.mdx',
  ]}
/>

<!-- prettier-ignore-end -->

<div class="aside">

> If want to learn more about writing your own addon and include some of these elements, read our <a href="./writing-addons">documentation</a>.

Author: Fixed with the latest commit.
